The compiling failed with openblas

Error:

```
[ 98%] Built target softmax_with_cross_entropy_op
/home/xxx/.third_party/install/openblas/lib/libopenblas.a(memory.o): In function `openblas_fork_handler':
memory.c:(.text+0x190): undefined reference to `pthread_atfork'
```

From this link https://github.com/xianyi/OpenBLAS/wiki/faq :

It seems that needs to link the pthread  library:

> How to solve undefined reference errors when statically linking against libopenblas.a

> On Linux, if OpenBLAS was compiled with threading support (USE_THREAD=1 by default), custom programs statically linked against libopenblas.a should also link to the pthread library e.g.:

>  gcc -static -I/opt/OpenBLAS/include -L/opt/OpenBLAS/lib -o my_program my_program.c -lopenblas -lpthread

--- a/cmake/generic.cmake
+++ b/cmake/generic.cmake
@@ -93,7 +93,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 if(NOT APPLE AND NOT ANDROID)
     find_package(Threads REQUIRED)
     link_libraries(${CMAKE_THREAD_LIBS_INIT})
-    set(CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE} -ldl -lrt")
+    set(CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE} -pthread -ldl -lrt")
 endif(NOT APPLE AND NOT ANDROID)
 
 function(merge_static_libs TARGET_NAME)

